api/proto/teleport/identitycenter/v1/service.proto

```diff
@@ -0,0 +1,126 @@
+// Copyright 2024 Gravitational, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package teleport.identitycenter.v1;
+
+import "teleport/identitycenter/v1/identitycenter.proto";
+import "teleport/provisioning/v1/provisioning.proto";
+
+option go_package = "github.com/gravitational/teleport/api/gen/proto/go/teleport/identitycenter/v1;identitycenterv1";
+
+// IdentityCenterService provides a remote-access API for the Identity Center
+// integration
+service IdentityCenterService {
+  // DescribePrincipal fetches summary information for a given principal
+  rpc DescribePrincipal(DescribePrincipalRequest) returns (PrincipalSummary);
+
+  // ListPrincipals returns paginated list of principal summaries
+  rpc ListPrincipals(ListPrincipalsRequest) returns (ListPrincipalsResponse);
+
+  // ResetPrincipal resets a principal's external ID and forces them to be re-
+  // provisioned into Identity Center
+  rpc ResetPrincipal(ResetPrincipalRequest) returns (ResetPrincipalResponse);
+}
+
+// PrincipalSummary summarizes the current state of a principal, combining
+// provisioning and account assignment info.
+// buf:lint:ignore PAGINATION_REQUIRED
+message PrincipalSummary {
+  // AssignmentSummary holds summary details about a principal's account
+  // assignment state
+  message AssignmentSummary {
+    ProvisioningState status = 1;
+    string external_id = 2;
+    string error = 3;
+  }
+
+  // ProvisioningSummary holds summary details about a principal's SCIM
+  // provisioning state assignment state
+  message ProvisioningSummary {
+    teleport.provisioning.v1.ProvisioningState status = 1;
+    string external_id = 2;
+    string error = 3;
+  }
```

Comment:
Nit: It took me a minute to rebuild the mental model of the integration and remember what the difference between assignment status and provisioning status is. Provisioning tracks users and groups created in the AWS IC, and assignment tracks account/permission set assignment to those, correct?

Can you update the godocs both here and below where you declare those fields to mention this explicitly?